pkg/hubble/parser/threefour/parser.go

```diff
@@ -316,13 +316,20 @@ func (p *Parser) resolveEndpoint(ip net.IP, datapathSecurityIdentity uint32) *pb
 	if p.endpointGetter != nil {
 		if ep, ok := p.endpointGetter.GetEndpointInfo(ip); ok {
 			epIdentity := resolveIdentityConflict(ep.GetIdentity())
-			return &pb.Endpoint{
+			e := &pb.Endpoint{
 				ID:        uint32(ep.GetID()),
 				Identity:  epIdentity,
 				Namespace: ep.GetK8sNamespace(),
 				Labels:    sortAndFilterLabels(p.log, ep.GetLabels(), epIdentity),
 				PodName:   ep.GetK8sPodName(),
 			}
+			if ep.GetPod() != nil {
+				if owners := ep.GetPod().GetOwnerReferences(); len(owners) != 0 {
+					e.WorkloadKind = owners[0].Kind
+					e.WorkloadName = owners[0].Name
+				}
```

Comment:
For cases where there are multiple owner references, it seems to me that `WorkloadKind` and `WorkloadName` might not be accurate. Shouldn't `WorkloadKind` and `WorkloadName` be string slices?

Reply:
For a Pod object, do we have the case where a pod has more than one owner?

Reply:
i think for pods there will be a single owner, but i think it makes sense to copy over the list anyways instead of picking the first item and letting the consumer of the API decide what to do in case there are multiple owners.

Reply:
OK.  That makes sense.